src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -1407,14 +1407,18 @@ void CheckOperationAttributes(ISymbol symbol, bool checkParents)
                             platformSpecificOperations.TryAdd(new KeyValuePair<IOperation, ISymbol>(operation, symbol), (notSuppressedAttributes, callSiteAttributes.Platforms));
                         }
                     }
-                    else if (TryCopyAttributesNotSuppressedByMsBuild(operationAttributes.Platforms!, msBuildPlatforms, out var copiedAttributes))
+                    else if (!OperationHasOnlyAssemblyAttributesAndCalledFromSameAssembly(operationAttributes, symbol, containingSymbol) &&
+                             TryCopyAttributesNotSuppressedByMsBuild(operationAttributes.Platforms!, msBuildPlatforms, out var copiedAttributes))
```

Comment:
I see this is for assembly only. What about the case where you have a partial type split in many files, and SupportedOSPlatform decorates the type, and that type has a nested type? Is that a case that would trigger a similar error?

FileOne.cs
```cs
[SupportedOSPlatform("...")]
public partial class MyPartialClass
{
    /// some code
}
```

FileTwo.cs
```cs
public partial class MyPartialClass
{
    /// some code
    public class MyNestedClass
    {
        [SupportedOsPlatformGuard("...")]
        public bool IsSupported => true;
    }
}
```

I don't know if this case would make sense, but I was just wondering if there were more cases to consider.

Reply:
I believe those split source will be parsed to one NamedSymbol, therefore there will be no any difference